modules/solid_mechanics/src/vectorpostprocessors/CrackFrontNormalStress.C

```diff
@@ -0,0 +1,159 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CrackFrontNormalStress.h"
+#include "Moose.h"
+#include "MooseEnum.h"
+#include "RankTwoTensor.h"
+#include "Conversion.h"
+#include "libmesh/string_to_enum.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/utility.h"
+#include "CrackFrontDefinition.h"
+#include "RankTwoScalarTools.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("SolidMechanicsApp", CrackFrontNormalStress);
+
+InputParameters
+CrackFrontNormalStress::validParams()
+{
+  InputParameters params = ElementVectorPostprocessor::validParams();
+  params.addRequiredParam<UserObjectName>("crack_front_definition",
+                                          "The CrackFrontDefinition user object name");
+  params.addRequiredParam<Real>("box_length", "Distance in front of crack front.");
+  params.addRequiredParam<Real>("box_height", "Distance normal to front of crack front.");
+  params.addParam<Real>("box_width", 1, "Distance tangent to front of crack front.");
```

Comment:
For 3D, it seems like it might be better to use the spacing of points along the crack front for the width. Let's discuss.

Reply:
I did not fix this in this pr.